src/main/java/org/jboss/xavier/integrations/rbac/RBACRouteBuilder.java

```diff
@@ -0,0 +1,149 @@
+                    try {
+                        xRHIdentityDecodedJsonNode = new ObjectMapper().reader().readTree(
+                                new String(
+                                        Base64.getDecoder().decode(xRHIdentity)
+                                )
+                        );
+                    } catch (IOException e) {
+                        Logger.getLogger(this.getClass().getName()).warning("Unable to read " + RBAC_X_RH_IDENTITY);
+                        throw new IllegalStateException("Could not read header " + RBAC_X_RH_IDENTITY);
+                    }
+                    exchange.getIn().setHeader(RBAC_X_RH_IDENTITY_DECODED, xRHIdentityDecodedJsonNode);
+                })
+                .process(exchange -> {
+                    // isOrgAdmin
+                    JsonNode xRHIdentity = exchange.getIn().getHeader(RBAC_X_RH_IDENTITY_DECODED, JsonNode.class);
+                    JsonNode isOrgAdmin = xRHIdentity.get("identity").get("user").get("is_org_admin");
+                    exchange.getIn().setHeader(RBAC_IS_ORG_ADMIN, isOrgAdmin.booleanValue());
+                })
+                .choice()
+                    .when(exchange -> exchange.getIn().getHeader(RBAC_IS_ORG_ADMIN, Boolean.class))
+                        .setHeader(RBAC_USER_ACCESS, constant(null))
+                    .endChoice()
+                    .otherwise()
+                        .enrich("direct:fetch-rbac-user-access", (oldExchange, newExchange) -> {
+                            List<Acl> acls = newExchange.getIn().getBody(List.class);
+                            Map<String, Map<String, List<String>>> accessForUser = RBACService.getAccessForUser(acls);
+
+                            oldExchange.getIn().setHeader(RBAC_USER_ACCESS, accessForUser);
+                            return oldExchange;
+                        })
+                    .endChoice()
+                .end();
+
+        from("direct:fetch-rbac-user-access")
+                .routeId("fetch-rbac-user-access")
+                .setHeader("access", ArrayList::new)
+                .setHeader("nextLink", () -> "")
+                .loopDoWhile(exchange -> exchange.getIn().getHeader("nextLink") != null)
+                    .setHeader(Exchange.HTTP_METHOD, constant(HttpMethods.GET))
+                    .setHeader(Exchange.CONTENT_TYPE, constant("application/json"))
+                    .setHeader(Exchange.HTTP_PATH, constant(rbacPath))
+                    .process(exchange -> {
+                        String httpQuery;
+                        String nextLink = exchange.getIn().getHeader("nextLink", String.class);
+                        int queryParamsIndex = nextLink.indexOf("?");
+                        if (queryParamsIndex != -1) {
+                            httpQuery = nextLink.substring(queryParamsIndex + 1);
+                        } else {
+                            httpQuery = "application=" + rbacApplicationName + "&limit=100&offset=0";
```

Comment:
It seems like `limit` query param is ignored from remote system since the response payload has the `1000` value
```
"first": "/api/rbac/v1/access/?application=migration-analytics&limit=1000&offset=0"
```
Could we remove it?
```suggestion
                            httpQuery = "application=" + rbacApplicationName + "&offset=0";
```
Other than this, since `rbacApplicationName` is coming from a property read at boot time and then never changed, i feel like this initial HTTP query could be created just once instead of in every call.
Something like defining inside `configure()` before `from("direct:fetch-rbac-user-access")`
```
final String initialHttpQuery = "application=" + rbacApplicationName + "&offset=0";
```
and then use it in the route.

Reply:
@mrizzi it seems that the default `Page Size` for that endpoint has been increased to `1000`, if my memory is working well, before it used to be just `10`. I suggest to keep the `limit` query parameter but increase it to `1000`. Keeping the `limit` query parameter will allow us to be in control of the `Page Size`, otherwise we will rely on default values that might change in the RBAC side (I don't think they will change it often but the possibility is still there). Wdyt?

Reply:
@carlosthe19916 My concern is exactly this one: we are in control at all since, right now, we have `100` and the RBAC response has `1000` totally discarding our request to just have `100` so i feel like this is not really a query param maybe on their side.
Since it's useless (we request `100` and it gave use `1000`) i think it should be removed since it gives the wrong perception of being able to control it.

Reply:
@mrizzi now I'm curious how you got the `1000` number even though you used `limit=100`, maybe you discovered some bug in my code or in the RBAC side... I've tried the OpenAPI interface and this is what i got:

![Screenshot from 2020-03-10 12-49-23](https://user-images.githubusercontent.com/2582866/76309782-c7e96180-62cd-11ea-8baa-38719109c708.png)

As you can see the rest endpoint is using `limit=88`

Reply:
Sorry @carlosthe19916, it has been a mistake i made in doing the remote call from terminal to simulate the RBAC calls.
Everything works as expected :+1: 
I think `limit=100` is fine